reascripts/common/libs/PathUtil.lua

```diff
@@ -0,0 +1,91 @@
+--[[
+
+  PathUtil.lua - Path utilities for ReaScript
+
+]]--
+
+PathUtil = {}
+
+-- Apply an extension to a file path, if it doesn't already have one.
+-- Empty or nil path returns the empty string.
+PathUtil.apply_extension = function(filepath, extension)
```

Comment:
I feel like it shouldn't be necessary to parse the path this much to add an extension.

I just searched for "python function to add extension to file if it does not have one already" and got this AI result:

```
def add_extension(filename, extension):
    """Adds the specified extension to the filename if it doesn't already have it."""

    root, ext = os.path.splitext(filename)
    if not ext:  # Check if the file has an extension
        return filename + extension
    else:
        return filename
```

`os.path.splitext()` accepts a full path without issue.

Reply:
I was more suggesting that we could implement something like `os.path.splitext()` in Lua. It works by finding the first dot starting from the end, sidestepping the need to parse the rest of the path: https://github.com/python/cpython/blob/3.10/Lib/genericpath.py#L114